relayer/crates/starknet-chain-components/src/impls/types/config.rs

```diff
@@ -13,6 +12,28 @@ pub struct StarknetRelayerConfig {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct StarknetChainConfig {
-    pub json_rpc_url: SocketAddr,
+    #[serde(with = "url_serde")]
```

Comment:
Don't the `url` crate implements `Serialize` for `Url` when the `serde` feature is enabled?

Reply:
Just to clarify, this `Url` is from `starknet::providers::Url`.

Reply:
That is just a re-export from `url`.

Reply:
Aah. I see 😅 I don't think, we can do much -- as the feature flag is not exposed via [`starknet` or `starknet-providers` feature flags](https://docs.rs/crate/starknet-providers/latest/source/Cargo.toml#36-37).

Reply:
We can import the `url` crate directly, and enable the feature there.

Reply:
I don't want to do this as they are not using the latest version. So, we have to pin it to Starknet's version. Maybe we can just move to `String`, if that's ok with you.

Reply:
Refactored using String: acd6759